### PR TITLE
Sftp Binding Component Docs

### PR DESCRIPTION
**Pull request by arslanmusta**
Thank you for helping make the Dapr documentation better!

**Please follow this checklist before submitting:**
- [x] Commits are signed with Developer Certificate of Origin (DCO - [learn more](https://docs.dapr.io/contributing/contributing-overview/#developer-certificate-of-origin-signing-your-work))
- [x] [Read the contribution guide](https://docs.dapr.io/contributing/docs-contrib/contributing-docs/)
- [x] Commands include options for Linux, MacOS, and Windows within codetabs
- [x] New file and folder names are globally unique
- [x] Page references use shortcodes instead of markdown or URL links
- [x] Images use HTML style and have alternative text
- [x] Places where multiple code/command options are given have codetabs

In addition, please fill out the following to help reviewers understand this pull request:

## Description

Adding Sftp Binding component documentation.

## Issue reference

Related PR: dapr/components-contrib#3505

### CODE REVIEW

**Review by hhunter-ms — APPROVED** [review]

File: daprdocs/content/en/reference/components-reference/supported-bindings/sftp.md [hunk 1]
```diff
@@ -0,0 +1,231 @@
+---
+type: docs
+title: "Sftp binding spec"
```
Comment (msfussell) [inline daprdocs/content/en/reference/components-reference/supported-bindings/sftp.md]:
```suggestion
title: "SFTP binding spec"
```

File: daprdocs/content/en/reference/components-reference/supported-bindings/sftp.md [hunk 2]
```diff
@@ -0,0 +1,231 @@
...
+linkTitle: "Sftp"
+description: "Detailed documentation on the Sftp binding component"
```
Comment (msfussell) [inline daprdocs/content/en/reference/components-reference/supported-bindings/sftp.md]:
```suggestion
description: "Detailed documentation on the Secure File Transfer Protocol (SFTP) binding component"
```

File: daprdocs/content/en/reference/components-reference/supported-bindings/sftp.md [hunk 3]
```diff
@@ -0,0 +1,231 @@
...
+  curl -d '{ "operation": "create", "data": "YOUR_BASE_64_CONTENT", "metadata": { "fileName": "my-test-file.jpg" } }' \
+        http://localhost:<dapr-port>/v1.0/bindings/<binding-name>
+  ```
+  {{% /codetab %}}
+
+{{< /tabs >}}
+
+#### Response
+
+The response body will contain the following JSON:
```
Comment (msfussell) [inline daprdocs/content/en/reference/components-reference/supported-bindings/sftp.md]:
```suggestion
The response body contains the following JSON:
```

File: daprdocs/content/en/reference/components-reference/supported-bindings/sftp.md [hunk 4]
```diff
@@ -0,0 +1,231 @@
...
+  curl -d '{ "operation": "delete", "metadata": { "fileName": "myfile" }}' \
+        http://localhost:<dapr-port>/v1.0/bindings/<binding-name>
+  ```
+  {{% /codetab %}}
+
+{{< /tabs >}}
+
+#### Response
+
+An HTTP 204 (No Content) and empty body will be returned if successful.
```
Comment (msfussell) [inline daprdocs/content/en/reference/components-reference/supported-bindings/sftp.md]:
```suggestion
An HTTP 204 (No Content) and empty body is returned if successful.
```

File: daprdocs/data/components/bindings/generic.yaml [hunk 5]
```diff
@@ -134,6 +134,14 @@
   features:
     input: true
     output: false
+- component: Sftp
```
Comment (msfussell) [inline daprdocs/data/components/bindings/generic.yaml]:
```suggestion
- component: SFTP
```

File: daprdocs/content/en/reference/components-reference/supported-bindings/sftp.md [hunk 6]
```diff
@@ -0,0 +1,231 @@
...
+title: "Sftp binding spec"
+linkTitle: "Sftp"
+description: "Detailed documentation on the Sftp binding component"
+aliases:
+  - "/operations/components/setup-bindings/supported-bindings/sftp/"
+---
+
+## Component format
+
+To set up the Sftp binding, create a component of type `bindings.sftp`. See this guide on how to create and apply a binding configuration.
```
Comment (msfussell) [inline daprdocs/content/en/reference/components-reference/supported-bindings/sftp.md]:
```suggestion
To set up the SMTP binding, create a component of type `bindings.sftp`. See this guide on how to create and apply a binding configuration.
```

**Review by msfussell — COMMENTED** [review]
@arslanmusta Looks great. Thanks for creating this. Generally keeping acronym to caps at the start of the doc with a full name description of the binding (similar to SMTP binding)

File: daprdocs/content/en/reference/components-reference/supported-bindings/sftp.md [hunk 7]
```diff
@@ -0,0 +1,231 @@
+---
+type: docs
+title: "SFTP binding spec"
+linkTitle: "Sftp"
```
Comment (hhunter-ms) [inline daprdocs/content/en/reference/components-reference/supported-bindings/sftp.md]:
```suggestion
linkTitle: "SFTP"
```

for consistency

File: daprdocs/content/en/reference/components-reference/supported-bindings/sftp.md [hunk 8]
```diff
@@ -0,0 +1,231 @@
...
+description: "Detailed documentation on the Secure File Transfer Protocol (SFTP) binding component"
+aliases:
+  - "/operations/components/setup-bindings/supported-bindings/sftp/"
+---
+
+## Component format
+
+To set up the SMTP binding, create a component of type `bindings.sftp`. See this guide on how to create and apply a binding configuration.
```
Comment (hhunter-ms) [inline daprdocs/content/en/reference/components-reference/supported-bindings/sftp.md]:
```suggestion
To set up the SFTP binding, create a component of type `bindings.sftp`. See [this guide]({{ ref bindings-overview.md }}) on how to create and apply a binding configuration.
```

File: daprdocs/content/en/reference/components-reference/supported-bindings/sftp.md [hunk 9]
```diff
@@ -0,0 +1,231 @@
...
+  - name: insecureIgnoreHostKey
+    value: "<bool>"
+```
+
+## Spec metadata fields
+
+| Field              | Required | Binding support |  Details | Example |
+|--------------------|:--------:|------------|-----|---------|
+| `rootPath` | Y | Output | Root path for default working directory | `"/path"` |
+| `address`             | Y        | Output |  Address of Sftp server | `"localhost:22"` |
```
Comment (hhunter-ms) [inline daprdocs/content/en/reference/components-reference/supported-bindings/sftp.md]:
```suggestion
| `address`             | Y        | Output |  Address of SFTP server | `"localhost:22"` |
```

File: daprdocs/content/en/reference/components-reference/supported-bindings/sftp.md [hunk 10]
```diff
@@ -0,0 +1,231 @@
...
+This component supports **output binding** with the following operations:
+
+- `create` : [Create file](#create-file)
+- `get` : [Get file](#get-file)
+- `list` : [List files](#list-files)
+- `delete` : [Delete file](#delete-file)
+
+### Create file
+
+To perform a create file operation, invoke the Sftp binding with a `POST` method and the following JSON body:
```
Comment (hhunter-ms) [inline daprdocs/content/en/reference/components-reference/supported-bindings/sftp.md]:
```suggestion
To perform a create file operation, invoke the SFTP binding with a `POST` method and the following JSON body:
```

File: daprdocs/content/en/reference/components-reference/supported-bindings/sftp.md [hunk 11]
```diff
@@ -0,0 +1,231 @@
...
+```json
+{
+   "fileName": "<filename>"
+}
+
+```
+
+### Get file
+
+To perform a get file operation, invoke the Sftp binding with a `POST` method and the following JSON body:
```
Comment (hhunter-ms) [inline daprdocs/content/en/reference/components-reference/supported-bindings/sftp.md]:
```suggestion
To perform a get file operation, invoke the SFTP binding with a `POST` method and the following JSON body:
```

File: daprdocs/content/en/reference/components-reference/supported-bindings/sftp.md [hunk 12]
```diff
@@ -0,0 +1,231 @@
...
+
+{{< /tabs >}}
+
+#### Response
+
+The response body contains the value stored in the file.
+
+### List files
+
+To perform a list files operation, invoke the Sftp binding with a `POST` method and the following JSON body:
```
Comment (hhunter-ms) [inline daprdocs/content/en/reference/components-reference/supported-bindings/sftp.md]:
```suggestion
To perform a list files operation, invoke the SFTP binding with a `POST` method and the following JSON body:
```

File: daprdocs/content/en/reference/components-reference/supported-bindings/sftp.md [hunk 13]
```diff
@@ -0,0 +1,231 @@
...
+
+{{< /tabs >}}
+
+#### Response
+
+The response is a JSON array of file names.
+
+### Delete file
+
+To perform a delete file operation, invoke the Sftp binding with a `POST` method and the following JSON body:
```
Comment (hhunter-ms) [inline daprdocs/content/en/reference/components-reference/supported-bindings/sftp.md]:
```suggestion
To perform a delete file operation, invoke the SFTP binding with a `POST` method and the following JSON body:
```

**Review by hhunter-ms — CHANGES_REQUESTED** [review]
quick consistency check

**Review by hhunter-ms — APPROVED** [review]
lgtm, thanks!

**Review by msfussell — APPROVED** [review]
LGTM

**Comment by marcduiker** [comment]
@holopin-bot @arslanmusta Thank you!

**Pull request merged**